contracts/proxies/SafeProxy.sol

```diff
@@ -45,9 +45,11 @@ contract SafeProxy {
                 // memory with a 12 byte offset from where the return data starts. Note that we **intentionally** only
                 // do this for the `masterCopy()` call, since the EVM `DELEGATECALL` opcode ignores the most-significant
                 // 12 bytes from the address, so we do not need to make sure the top bytes are cleared when proxying
-                // calls to the `singleton`. This saves us a tiny amount of gas per proxied call.
-                mstore(0x0c, shl(96, _singleton))
-                return(0, 0x20)
+                // calls to the `singleton`. This saves us a tiny amount of gas per proxied call. Additionally, we write
+                // to the "zero-memory" slot, this guarantees that there are no dirty bits in the first 12 bytes in the
```

Comment:
If the goal of this change is to align with the solidity compiler, I'm not sure if writing to the zero memory slot is the best approach. Quote from the docs:

> The zero slot is used as initial value for dynamic memory arrays and should never be written to (the free memory pointer points to 0x80 initially).

Other than that, I'm good with the change

Reply:
Good point... maybe we should just properly clear the bits...

Reply:
I thought a little more about this and I think taking advantage of the 0 slot makes the most sense here:

- We already don't declare the memory as "safe" and we never return control back to Solidity - so messing with its memory layout is not an issue
- the `calldatacopy(0, 0, calldatasize())` also writes over the 0-memory slot and potentially sets to a non-zero value, so we aren't disrespecting the memory any more than we already are
- Fully clearing the bits adds more code, and (IMO) for no benefit (since we won't be respecting memory safety rules any more than we currently are).